File: cmstestsuite/tasks/communication_many_stdio_stubbed/code/stub.pas
program stub;
uses sysutils, user1, user2;

var
    n: longint;
    procid: longint;

begin
    procid := StrToInt(ParamStr(1));

    while True do
    begin
        readln(n);
        if (n = 0) then
            break;
        if (procid = 0) then
            writeln('correct ', userfunc1(n))
        else
            writeln('correct ', userfunc2(n));
        flush(output);
    end;
end.

File: cmstestsuite/tasks/communication_many_stdio_stubbed/code/stub.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class stub {

    public static void main(String[] args) throws IOException {
        int procid = Integer.parseInt(args[0]);

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        while (true) {
            int n = Integer.parseInt(br.readLine());
            if (n == 0)
                break;
            if (procid == 0)
                System.out.println("correct " + user1.userfunc1(n));
            else
                System.out.println("correct " + user2.userfunc2(n));
            System.out.flush();
        }
    }
}
